homeassistant/components/group/binary_sensor.py

```diff
@@ -0,0 +1,130 @@
+"""This platform allows several binary sensor to be grouped into one binary sensor."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASSES_SCHEMA,
+    DOMAIN as BINARY_SENSOR_DOMAIN,
+    PLATFORM_SCHEMA,
+    BinarySensorEntity,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_DEVICE_CLASS,
+    CONF_ENTITIES,
+    CONF_NAME,
+    CONF_UNIQUE_ID,
+    STATE_ON,
+    STATE_UNAVAILABLE,
+)
+from homeassistant.core import CoreState, Event, HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_track_state_change_event
+from homeassistant.helpers.typing import ConfigType
+
+from . import GroupEntity
+
+DEFAULT_NAME = "Binary Sensor Group"
+
+CONF_ALL = "all"
+REG_KEY = f"{BINARY_SENSOR_DOMAIN}_registry"
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_ENTITIES): cv.entities_domain(BINARY_SENSOR_DOMAIN),
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_UNIQUE_ID): cv.string,
+        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
+        vol.Optional(CONF_ALL): cv.boolean,
+    }
+)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: dict[str, Any] | None = None,
+) -> None:
+    """Set up the Group Binary Sensor platform."""
+    async_add_entities(
+        [
+            BinarySensorGroup(
+                config.get(CONF_UNIQUE_ID),
+                config[CONF_NAME],
+                config.get(CONF_DEVICE_CLASS),
+                config[CONF_ENTITIES],
+                config.get(CONF_ALL),
+            )
+        ]
+    )
+
+
+class BinarySensorGroup(GroupEntity, BinarySensorEntity):
+    """Representation of a BinarySensorGroup."""
+
+    _attr_assumed_state: bool = True
+
+    def __init__(
+        self,
+        unique_id: str | None,
+        name: str,
+        device_class: str | None,
+        entity_ids: list[str],
+        mode: str | None,
+    ) -> None:
+        """Initialize a BinarySensorGroup entity."""
+        super().__init__()
+        self._entity_ids = entity_ids
+        self._attr_name = name
+        self._attr_extra_state_attributes = {ATTR_ENTITY_ID: entity_ids}
+        self._attr_unique_id = unique_id
+        self._device_class = device_class
+        self._state: str | None = None
+        self.mode = any
+        if mode:
+            self.mode = all
```

Comment:
This is already done in `GroupEntity`

Reply:
OK, I was trying to call GroupEntity's __init__ method but flake8 kept complaining. Maybe the super is getting called, but it seemed this method was needed after I removed the explicit call to init.

Reply:
My mistake, I confused `GroupEntity` and `Group`, sorry about that.